### PR TITLE
chore: migrate to aws-sdk v3

### PR DESCRIPTION
**Pull request by jsetton**
*Description of changes:*

This change migrates aws-sdk support from v2 to v3. The former is going into maintenance mode sometimes this year. This will remove the below warning from being generated during the skill infrastructure deployment.

```
==================== Deploy Skill Infrastructure ====================
(node:927) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
```

It includes some minor code refactoring and the removal of a duplicate test file.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/builtins/deploy-delegates/lambda-deployer/helper.js [hunk 1]
```diff
@@ -67,67 +71,71 @@ To ignore this error run "ask deploy --ignore-hash".`);
...
-        } else {
-          callback(`Failed to retrieve IAM role (${roleArn}) for Lambda. ${roleErr}`);
-        }
+    try {
+      const role = await iamClient.getIAMRole(roleArn);
+      roleArn = role.Arn;
+      reporter.updateStatus(`Current IAM role: "${roleArn}"`);
+      callback(null, roleArn);
+    } catch (err) {
+      if (err.$metadata.httpStatusCode === 404) {
```
Comment (doiron) [inline lib/builtins/deploy-delegates/lambda-deployer/helper.js]: nit could err or err.$metadata be null here?

File: lib/clients/aws-client/iam-client.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const command = new CreateRoleCommand({
+      RoleName: roleName,
+      AssumeRolePolicyDocument: JSON.stringify(policy),
+    });
+    return this.client.send(command).then((data: CreateRoleCommandOutput) => data.Role);
+  }
+
+  /**
+   * Attaches the specified managed policy to the specified IAM role.
+   * @param roleArn The Amazon Resource Name (ARN) specifying the group.
```
Comment (doiron) [inline lib/clients/aws-client/iam-client.ts]: nit:  is it a typo `group`.  Should this not be `role`? 

same on line 71

File: lib/clients/aws-client/lambda-client.ts [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      credentials: this.credentials,
+      region: this.region,
+    });
+  }
+
+  /**
+   * Creates a lambda function
+   * @param functionName The name of the lambda function
+   * @param config The configuration of the lambda function
+   * @param role The arm of the iam role
```
Comment (doiron) [inline lib/clients/aws-client/lambda-client.ts]: nit: typo?  `arm`   should this be `arn`?

**Review by doiron — APPROVED** [review]
wow this is really great! My entire team here is very grateful for your time / efforts on getting this done! 

thank you! thank you! thank you!

**Comment by doiron** [comment]
ignore the bad merge.  fixing it.

**Comment by jsetton** [comment]
Do you want me to resolve the review comments you added?

**Comment by doiron** [comment]
no I'm about done testing it right now.   sorry about that was attempting to merge the package-lock.json file manually using the browser   lol  bad idea..

**Comment by jsetton** [comment]
I fixed the conflicts and applied the changes based on your review.

**Comment by jsetton** [comment]
I just realized you created #458.

**Review by doiron — APPROVED** [review]

**Comment by doiron** [comment]
thanks for fixing the conflict and updating,  I got side tracked trying to fix some of the tests that are failing.   I'm seeing functional test failures with these changes locally  so currently trying to debug what is going on there.  Once I can understand what needs to be done to fix them, I'll merge.
thanks again this PR is a enormous help! 

update: oh one more thing, if you rebase now the integration tests should hopefully work.

**Comment by doiron** [comment]
functional tests are failing, but I have a local fix for it, will push a PR after this bigger change is merged.

**Pull request merged**